quinn/src/endpoint.rs

```diff
@@ -253,39 +253,43 @@ where
 {
     fn drive_recv(&mut self, cx: &mut Context, now: Instant) -> Result<bool, io::Error> {
         let mut recvd = 0;
+        let mut metas = [RecvMeta::default(); BATCH_SIZE];
+        let mut iovs = [IoSliceMut::new(&mut [])];
+        for (iov, buf) in iovs
+            .iter_mut()
+            .zip(self.recv_buf.chunks_mut(self.recv_buf.len() / BATCH_SIZE))
+        {
+            *iov = IoSliceMut::new(buf);
+        }
         loop {
-            let mut meta = [RecvMeta::default()];
-            match self
-                .socket
-                .poll_recv(cx, &mut [IoSliceMut::new(&mut self.recv_buf)], &mut meta)
-            {
+            let result = self.socket.poll_recv(cx, &mut iovs, &mut metas);
+            match result {
```

Comment:
We can match on the expression directly, right?

Reply:
Right; stray change from a previous iteration. Fixed.